### PR TITLE
Validate string literal generation vs "naughty strings"

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @srawlins – thanks for the inspiration!

**Comment by srawlins** [comment]
Haha nice! No new test failures?

**Comment by kevmoo** [comment]
> Haha nice! No new test failures?

Thankfully not, but given the complexity of what I'm pulling off, it's nice to have some test cases that I didn't write.

File: json_serializable/test/json_literal_test.dart [hunk 1]
```diff
@@ -25,4 +25,15 @@ main() {
 
     expect(dartString, dataString);
   });
+
+  test('naught strings', () {
```
Comment (natebosch) [inline json_serializable/test/json_literal_test.dart]: typo?

File: json_serializable/test/test_files/big-list-of-naughty-strings.json [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+ "classic",
+ "Tyson Gay",
+ "Dick Van Dyke",
+ "basement",
+ "If you're reading this, you've been in a coma for almost 20 years now. We're trying a new technique. We don't know where this message will end up in your dream, but we hope it works. Please wake up, we miss you.",
+ "Roses are \u001b[0;31mred\u001b[0m, violets are \u001b[0;34mblue. Hope you enjoy terminal hue",
+ "But now...\u001b[20Cfor my greatest trick...\u001b[8m",
+ "The quic\b\b\b\b\b\bk brown fo\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007x... [Beeeep]",
+ "Powerلُلُصّبُلُلصّبُررً ॣ ॣh ॣ ॣ冗"
+]
```
Comment (natebosch) [inline json_serializable/test/test_files/big-list-of-naughty-strings.json]: [nit] Add a newline?

Comment (kevmoo) [reply json_serializable/test/test_files/big-list-of-naughty-strings.json]: done

File: json_serializable/test/json_literal_test.dart (hunk 1 above)
Comment (kevmoo) [reply json_serializable/test/json_literal_test.dart]: yup

**Pull request merged**